atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraRepairEteTest.java

```diff
@@ -0,0 +1,164 @@
+                cassandraRepairHelper.getRangesToRepair(Namespace.of(NAMESPACE), TABLE_1);
+
+        KeyedStream.stream(rangesToRepair)
+                .forEach((address, cqlRangesForHost) ->
+                        assertRangesToRepairAreSubsetsOfRangesFromTokenMap(fullTokenMap, address, cqlRangesForHost));
+    }
+
+    // The ranges in CQL should be a subset of the Thrift ranges, except that the CQL ranges are also snipped,
+    // such that if the thrift range is [5..9] but we don't have data after 7, then the CQL range will be [5..7]
+    private void assertRangesToRepairAreSubsetsOfRangesFromTokenMap(
+            Map<InetSocketAddress, Set<LightweightOppTokenRange>> fullTokenMap,
+            InetSocketAddress address,
+            Set<LightweightOppTokenRange> cqlRangesForHost) {
+        String hostName = address.getHostName();
+        InetSocketAddress thriftAddr = new InetSocketAddress(hostName, MultiCassandraUtils.CASSANDRA_THRIFT_PORT);
+        Set<LightweightOppTokenRange> thriftRangesForHost = fullTokenMap.get(thriftAddr);
+        assertThat(thriftRangesForHost).isNotNull();
+        cqlRangesForHost.forEach(range -> assertThat(thriftRangesForHost.stream()
+                        .anyMatch(thriftRange -> thriftRange.left().equals(range.left())))
+                .isTrue());
+    }
+
+    private Map<InetSocketAddress, Set<LightweightOppTokenRange>> getFullTokenMap() {
+        CassandraService cassandraService = CassandraService.createInitialised(
+                MetricsManagers.createForTests(),
+                config,
+                new Blacklist(config),
+                new CassandraClientPoolMetrics(MetricsManagers.createForTests()));
+        return invert(cassandraService.getTokenMap());
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private Map<InetSocketAddress, Set<LightweightOppTokenRange>> invert(
+            RangeMap<LightweightOppToken, List<InetSocketAddress>> tokenMap) {
+        Map<InetSocketAddress, Set<LightweightOppTokenRange>> invertedMap = new HashMap<>();
+        tokenMap.asMapOfRanges()
+                .forEach((range, addresses) -> addresses.forEach(addr -> {
+                    Set<LightweightOppTokenRange> existingRanges = invertedMap.getOrDefault(addr, new HashSet<>());
+                    existingRanges.add(toTokenRange(range));
+                    invertedMap.put(addr, existingRanges);
+                }));
+
+        return invertedMap;
+    }
+
+    private LightweightOppTokenRange toTokenRange(Range<LightweightOppToken> range) {
+        LightweightOppTokenRange.Builder rangeBuilder = LightweightOppTokenRange.builder();
+        if (range.hasLowerBound()) {
+            rangeBuilder.left(range.lowerEndpoint());
+        }
+        if (range.hasUpperBound()) {
+            rangeBuilder.right(range.upperEndpoint());
+        }
+        return rangeBuilder.build();
+    }
+
+    private String stringify(Map<InetSocketAddress, Set<LightweightOppTokenRange>> ranges) {
```

Comment:
This is probably a leftover from testing?

Reply:
yep, unused -- I'll remove